lib/schema.js

```diff
@@ -302,34 +325,57 @@ class Metaschema {
     }
   }
 
-  // Create category instance
-  //   def - <Object> | <string> field definition
+  // Create domain instance
+  //   def - <Object> field definition
   //   value - <any> value to check/create
   // Returns: instance of `def`
-  createInstance(def, value) {
-    let name;
-    if (typeof def === 'string') name = def;
-    else name = def.category || def.domain;
-
-    const decorator = extractDecorator(def);
-    const category = this.categories.get(name);
-    if (category && category.factory) {
-      return category.factory(value);
-    }
+  createDomainInstance(def, value) {
+    if (def.domain === 'Id') return new Uint64(value);
+    if (def.domain === 'Money') return new Money(value);
+    const domain = this.domains.get(def.domain);
+    const decorator = extractDecorator(domain);
+    if (domain.normalize) value = domain.normalize(value, null, domain);
+    if (domain.parse) value = domain.parse(value);
     if (decorator === 'List') {
-      const cat = this.categories.get(def.category);
+      const cat = this.categories.get(domain.category);
       if (!cat) return null;
       const checked = value.map(v => this.createInstance(cat, v));
       if (checked.some(v => v === null)) return null;
       return checked;
     }
-    const domain = this.domains.get(name);
+    if (decorator === 'Enum') {
+      const valid = domain.Class.has(value);
+      return valid ? domain.Class.from(value) : null;
+    }
+    if (decorator === 'Flags') {
+      if (typeof value === 'string') {
+        return domain.Class.from(new Uint64(value));
+      }
+      const valid =
+        Array.isArray(value) &&
+        value.every(v => domain.Class.has(v));
+      return valid ? domain.Class.from(...value) : null;
+    }
     const type = typeof value;
-    const expected = domain ? domain.type : name;
+    const expected = domain.type;
+    if (expected === 'bigint') return new Int64(value);
+    if (expected === 'object') {
+      const classWrapper = classWrappers[domain.class];
+      return classWrapper ? classWrapper(value) : null;
+    }
     if (type === expected) return value;
     return null;
   }
 
+  // Create category instance
+  //   def - <Object> field definition
```

Comment:
This should be just a string.

Reply:
In this case there is no need in `createInstance` for category, because we have `buildCategory` which does absolutely the same

Reply:
We can delete `buildCategory` in that case.